### PR TITLE
Need to specify the MySQL driver for HikariCP to work

### PR DESCRIPTION
**Pull request by rtitle**
... and stop leaking `ActorSystem`s in tests. No JIRA.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/DbReference.scala [hunk 1]
```diff
@@ -46,13 +45,9 @@ object DbReference extends LazyLogging {
     }
   }
 
-  def init(config: Config, system: ActorSystem)(implicit executionContext: ExecutionContext): DbReference = {
+  def init(config: Config)(implicit executionContext: ExecutionContext): DbReference = {
     val dbConfig = DatabaseConfig.forConfig[JdbcProfile]("slick", config)
 
-    system.registerOnTermination {
```
Comment (rtitle) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/DbReference.scala]: This was the only reason `DbReference` needed an `ActorSystem` - I just moved it to `Boot.scala`.

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoServiceSpec.scala [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, FlatSpecLike, Matchers}
 
+class LeonardoServiceSpec extends TestKit(ActorSystem("leonardotest")) with FlatSpecLike with Matchers with BeforeAndAfterAll {
+  import system.dispatcher
 
-class LeonardoServiceSpec extends FlatSpec with Matchers {
-  import scala.concurrent.ExecutionContext.Implicits.global
-
-  implicit val system = ActorSystem("leonardotest")
+  override def afterAll(): Unit = {
```
Comment (rtitle) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoServiceSpec.scala]: Akka `TestKit` wants you to clean up the `ActorSystem` yourself; `ScalatestRouteTest` does it for you. Go figure.

File: src/test/resources/reference.conf [hunk 3]
```diff
@@ -15,6 +15,7 @@ slick {
   port = 3306
   port = ${?MYSQL_PORT}
   db {
+    driver = "com.mysql.cj.jdbc.Driver"
```
Comment (rtitle) [inline src/test/resources/reference.conf]: HikariCP gets very confused without this

Comment (jmthibault79) [inline src/test/resources/reference.conf]: Huh.  Good catch!  We'll need to add this to firecloud-dev too

Comment (rtitle) [inline src/test/resources/reference.conf]: Yeah. Speaking of, I couldn't find a Leo config in firecloud-develop. Is it not merged yet?

https://github.com/broadinstitute/firecloud-develop/tree/dev/configs

BTW, rawls.conf has this setting: https://github.com/broadinstitute/firecloud-develop/blob/dev/configs/rawls/rawls.conf.ctmpl#L78. (Apparently `com.mysql.jdbc.Driver` is deprecated now in favor of `com.mysql.cj.jdbc.Driver`.)

Comment (jmthibault79) [inline src/test/resources/reference.conf]: It's on the leonardo-configs branch for now, to help ensure that it doesn't go to prod.

Rawls uses older versions of Slick, Mysql, and Liquibase, so the format is slightly different.

**Pull request merged**